src/main/java/org/cyclonedx/generators/xml/AbstractBomXmlGenerator.java

```diff
@@ -79,413 +89,36 @@ public DocumentBuilder buildSecureDocumentBuilder() throws ParserConfigurationEx
-                    }
-                }
-            }
-        }
-    }
+            this.doc = docBuilder.parse(
+                new InputSource(
+                    new StringReader(
+                        toXML(bom))));
 
-    Element createRootElement(String name, Object value, Attribute... attributes) {
-        Element node = null;
-        if (value != null || attributes.length > 0) {
-            node = doc.createElementNS(getSchemaVersion().getNamespace(), name);
-            for (Attribute attribute: attributes) {
-                node.setAttribute(attribute.getKey(), attribute.getValue());
-            }
-            if (value != null) {
-                node.appendChild(doc.createTextNode(value.toString()));
-            }
-            doc.appendChild(node);
-        }
-        return node;
-    }
+            this.doc.setXmlStandalone(true);
 
-    static String stripBreaks(String in) {
-        if (in == null) {
-            return null;
+            return this.doc;
+        } catch (SAXException | ParserConfigurationException | IOException | GeneratorException ex) {
+            throw new ParserConfigurationException(ex.toString());
         }
-        return in.trim().replace("\r\n", " ").replace("\n", " ").replace("\t", " ").replace("\r", " ").replaceAll(" +", " ");
     }
 
-    /**
-     * Creates a text representation of a CycloneDX BoM Document.
-     * @return a String of the BoM
-     * @throws TransformerException an TransformerException
-     * @since 1.1.0
-     */
-    public String toXmlString() throws TransformerException {
-        if (doc == null) {
-            return null;
+    public String toXML(final Bom bom) throws GeneratorException {
+        try {
+            return PROLOG + mapper.writeValueAsString(bom);
+        } catch (JsonProcessingException ex) {
+            throw new GeneratorException(ex);
         }
-        final DOMSource domSource = new DOMSource(doc);
-        final StringWriter writer = new StringWriter();
-        final StreamResult result = new StreamResult(writer);
-        final TransformerFactory tf = TransformerFactory.newInstance();
-        tf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
-        final Transformer transformer = tf.newTransformer();
-        transformer.setOutputProperty(OutputKeys.ENCODING, StandardCharsets.UTF_8.name());
-        transformer.setOutputProperty(OutputKeys.INDENT, "yes");
-        transformer.setOutputProperty(OutputKeys.DOCTYPE_PUBLIC, "yes");
-        transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "4");
-        transformer.transform(domSource, result);
-
-        return writer.toString();
     }
 
     /**
      * Creates a text representation of a CycloneDX BoM Document. This method
-     * calls {@link #toXmlString()} and will return null if {@link #toXmlString()}
+     * calls {@link #toXmlString()} and will return an empty string if {@link #toXmlString()}
```

Comment:
What's the idea of changing this behavior? should we return null?

Reply:
It's a good question, I personally prefer returning empty strings ON a string type, exclusively, over null. It makes handling easier on the other end. I know that people think of null as classically the absence of data, but with strings it just makes handling things kinda dumb. WDYT?

Reply:
End consumption of something like this: `if (toXmlString() != null && toXmlString() != "")` etc..., I think in a case like this, null and empty string more or less represent the same thing (although not necessarily programmatically, but you know, my preference is to make consumption by a user easier).